common/djangoapps/third_party_auth/saml.py

```diff
@@ -113,107 +112,87 @@ class SapSuccessFactorsIdentityProvider(SAMLIdentityProvider):
-                'private_key': self.sapsf_private_key,
+                'token_url': idp.conf['sapsf_oauth_root_url'] + 'token',
+                'private_key': idp.conf['sapsf_private_key'],
             },
-            timeout=10,
+            timeout=timeout,
         )
+        log.warning('Posted assertion')
         assertion.raise_for_status()
         assertion = assertion.text
         token = session.post(
-            self.sapsf_token_url,
+            idp.conf['sapsf_oauth_root_url'] + 'token',
             data={
-                'client_id': self.odata_client_id,
-                'company_id': self.odata_company_id,
+                'client_id': idp.conf['odata_client_id'],
+                'company_id': idp.conf['odata_company_id'],
                 'grant_type': 'urn:ietf:params:oauth:grant-type:saml2-bearer',
                 'assertion': assertion,
             },
-            timeout=10,
+            timeout=timeout,
         )
+        log.warning('Posted and got token')
         token.raise_for_status()
         token = token.json()['access_token']
         session.headers.update({'Authorization': 'Bearer {}'.format(token), 'Accept': 'application/json'})
         return session
 
-    def get_user_details(self, attributes):
+    def get_user_details(self, response):
         """
         Attempt to get rich user details from the SAP SuccessFactors OData API. If we're missing any
         of the details we need to do that, fail nicely by returning the details we're able to extract
         from just the SAML response and log a warning.
         """
-        details = super(SapSuccessFactorsIdentityProvider, self).get_user_details(attributes)
-        if self.missing_variables():
+        details = super(SapSuccessFactorsAuthBackend, self).get_user_details(response)
+        idp = self.get_idp(response['idp_name'])
+        if self.missing_variables(idp):
             # If there aren't enough details to make the request, log a warning and return the details
             # from the SAML assertion.
             return details
         username = details['username']
+        odata_timeout = int(idp.conf.get('odata_api_timeout_interval', 20))
         try:
-            client = self.get_odata_api_client(user_id=username)
+            client = self.get_odata_api_client(idp, user_id=username, timeout=odata_timeout)
             response = client.get(
                 '{root_url}User(userId=\'{user_id}\')?$select=username,firstName,lastName,defaultFullName,email'.format(
-                    root_url=self.odata_api_root_url,
+                    root_url=idp.conf['odata_api_root_url'],
                     user_id=username
                 ),
-                timeout=10,
+                timeout=odata_timeout,
             )
             response.raise_for_status()
             response = response.json()
-        except requests.RequestException:
+        except requests.RequestException as err:
```

Comment:
Ah. Yeah, I have a bunch of debugging stuff that I actually need to go through and excise. Whoops!